protocol/x/accountplus/ante/circuit_breaker.go

```diff
@@ -0,0 +1,46 @@
+package ante
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/dydxprotocol/v4-chain/protocol/x/accountplus/keeper"
+	"github.com/dydxprotocol/v4-chain/protocol/x/accountplus/lib"
+)
+
+// CircuitBreakerDecorator routes transactions through appropriate ante handlers based on
+// the IsCircuitBreakActive function.
+type CircuitBreakerDecorator struct {
+	accountPlusKeeper            *keeper.Keeper
+	authenticatorAnteHandlerFlow sdk.AnteHandler
+	originalAnteHandlerFlow      sdk.AnteHandler
+}
+
+// NewCircuitBreakerDecorator creates a new instance of CircuitBreakerDecorator with the provided parameters.
+func NewCircuitBreakerDecorator(
+	accountPlusKeeper *keeper.Keeper,
+	auth sdk.AnteHandler,
+	classic sdk.AnteHandler,
+) CircuitBreakerDecorator {
+	return CircuitBreakerDecorator{
+		accountPlusKeeper:            accountPlusKeeper,
+		authenticatorAnteHandlerFlow: auth,
+		originalAnteHandlerFlow:      classic,
+	}
+}
+
+// AnteHandle checks if a tx is a smart account tx and routes it through the correct series of ante handlers.
+func (ad CircuitBreakerDecorator) AnteHandle(
+	ctx sdk.Context,
+	tx sdk.Tx,
+	simulate bool,
+	next sdk.AnteHandler,
+) (newCtx sdk.Context, err error) {
+	// Check that the authenticator flow is active
+	if specified, _ := lib.HasSelectedAuthenticatorTxExtensionSpecified(tx, ad.accountPlusKeeper); specified {
+		// Return and call the AnteHandle function on all the authenticator decorators.
+		return ad.authenticatorAnteHandlerFlow(ctx, tx, simulate)
+	}
+
+	// Return and call the AnteHandle function on all the original decorators.
+	return ad.originalAnteHandlerFlow(ctx, tx, simulate)
+}
```

Comment:
Note for @teddyding - This is different from Osmosis's logic. 

Osmosis
- If smart account active AND tx extension present -> use new auth flow
- else fallback to default flow 

dYdX
- if tx extension missing -> use original flow 
- else use new flow 
-- *if smart account inactive -> return error* (this is the gist of the diff, no longer falls back to original flow but returns error instead)
-- proceed with new flow

Reply:
Reason to internalize this is for the resting maker order issue. Since the proposed solution in the slack channel doesn't actually run ante handlers, we need to internalize so that we no longer need to worry about 
- order placed using new auth flow 
- gov turns off smart account 
- order now falling back to old flow

Reply:
Makes sense! It may be helpful to document this (why we aren't conditioning based on `IsSmartAcccountActive` here - the restring maker order issue is not obvious